### PR TITLE
Use Forkchoice Store To Build Attestation

### PR DESCRIPTION
**Pull request by terencechain**
This PR integrates the forkchoice store for building attestations, resulting in two significant changes:

- Removal of the Head State Dependency: This update eliminates the need for using the head state to build attestations. Consequently, it reduces much of the boilerplate code, including the processing of attestation up to a given slot, among other things.
- Deprecation of the Attestation Cache: Given that forkchoice store retrieval is comparably fast, if not faster, the attestation cache becomes redundant. Removing it will also help in freeing up memory resources.

### CODE REVIEW

File: beacon-chain/cache/committee.go [hunk 1]
```diff
@@ -23,6 +23,11 @@ const (
 	// maxCommitteesCacheSize defines the max number of shuffled committees on per randao basis can cache.
 	// Due to reorgs and long finality, it's good to keep the old cache around for quickly switch over.
 	maxCommitteesCacheSize = int(32)
+
+	// delay parameters for checking in progress requests.
+	minDelay    = float64(10)        // 10 nanoseconds
+	maxDelay    = float64(100000000) // 0.1 second
+	delayFactor = 1.1
```
Comment (terencechain) [inline beacon-chain/cache/committee.go:30]: These configs were shared between attestation data cache and committee cache. Now that attestation data cache was deleted, we are moving the configs here

File: beacon-chain/cache/skip_slot_cache.go [hunk 2]
```diff
@@ -14,6 +14,13 @@ import (
 	"go.opencensus.io/trace"
 )
 
+const (
+	// delay parameters for checking in progress requests.
+	minDelay    = float64(10)        // 10 nanoseconds
+	maxDelay    = float64(100000000) // 0.1 second
+	delayFactor = 1.1
+)
+
```
Comment (terencechain) [inline beacon-chain/cache/skip_slot_cache.go:23]: These were defined in `attestation_data.go` but shared with skip slot cache. Moving it here now

File: beacon-chain/rpc/eth/validator/handlers_test.go [hunk 3]
```diff
@@ -974,97 +962,13 @@ func TestGetAttestationData(t *testing.T) {
 		assert.Equal(t, http.StatusOK, writer.Code)
 	})
 
-	t.Run("handles in progress request", func(t *testing.T) {
```
Comment (potuz) [inline beacon-chain/rpc/eth/validator/handlers_test.go:977]: why are these tests removed?

Comment (terencechain) [reply beacon-chain/rpc/eth/validator/handlers_test.go:977]: The notion of in-progress was part of the attestation data cache's mark-in-progress function, now that attestation data is removed and no longer used, this test becomes irrelevant

**Pull request closed without merging**